singularity-eos/eos/modifiers/ramps_eos.hpp

```diff
@@ -0,0 +1,239 @@
+//------------------------------------------------------------------------------
+// © 2021-2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_EOS_RAMPS_EOS_
+#define _SINGULARITY_EOS_EOS_RAMPS_EOS_
+
+#include "stdio.h"
+#include <cassert>
+#include <cmath>
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+#include <utility>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/base/constants.hpp>
+#include <singularity-eos/base/eos_error.hpp>
+#include <singularity-eos/eos/eos_base.hpp>
+
+namespace singularity {
+
+using namespace eos_base;
+
+template <typename T>
+void pAlpha2SAPRampParams(const T &eos, const Real alpha0, const Real Pe, const Real Pc,
+                          Real &r0, Real &a, Real &b, Real &c) {
+  // get reference conditions
+  Real rho0, T0, sie0, P0, cv0, bmod0, dpde0, dvdt0;
+  Real rmid, r1;
+  eos.ValuesAtReferenceState(rho0, T0, sie0, P0, cv0, bmod0, dpde0, dvdt0);
```

Comment:
This is a conversation for a later time---but with the new PTE solver, do we *need* `ValuesAtReferenceState`? If not, perhaps we should deprecate it and remove it.

Reply:
We at least need the density at reference state in order for us to initialize the parameters from p-alpha parameters.